code/ship/ship.cpp

```diff
@@ -11676,6 +11678,14 @@ int ship_fire_primary(object * obj, int force, bool rollback_shot)
 					points = num_slots;
 				}
 
+				// The player is trying to fire primaries which are locked. We've set the Fail_sound_locked_primaries flag, so it should make the fail sound.
+				if  (obj == Player_obj && shipp->flags[Ship_Flags::Primaries_locked] && shipp->flags[Ship_Flags::Fail_sound_locked_primary])
+				{					
+					ship_maybe_do_primary_fail_sound_hud(false);
+					ship_stop_fire_primary_bank(obj, bank_to_fire);
+					continue;
+				}
+
```

Comment:
This seems a bit far down the function to do this; I know this is adjacent to the regular `ship_maybe_do_primary_fail_sound_hud()` call, but that has reasons to be in the nitty-gritty of this function, while this kinda doesn't; it should have the same effect if you called `ship_maybe_do_primary_fail_sound_hud(false)` around line 11302 immediately before returning.
```suggestion
```